crates/wasm-metadata/src/lib.rs

```diff
@@ -291,6 +313,19 @@ fn rewrite_wasm(
                 names.section()?.as_custom().append_to(&mut output);
             }
 
+            CustomSection(c) if c.name() == "registry-metadata" && stack.len() == 0 => {
+                // Pass section through if a new registry metadata isn't provided, otherwise ignore and overwrite with new
```

Comment:
Should this perhaps be an error if metadata is already in the file and metadata is also being added?

Reply:
I don't know to be honest. It depends on if we want the ability to overwrite it or not I guess? I'm happy to change it to an error if we don't want the ability to overwrite registry-metadata.

I originally had it implemented where you could add new registry metadata by merging it with the existing metadata but it became a bit weird if you changed the license expression, for example if you had a custom license that is referenced in the expression. Then you change the license expression to not include the custom license, then you would also need to remove the old license. What would you do in that case, just set it to an empty array in the input JSON if using the cli (not the best user experience)? Or would we remove it automatically and silently handle it for the user (not what I would want or expect)?

Reply:
After discussing with @silesmo, I think overwrite makes sense for `registry-metadata`. Feels like the expected behavior for this.

Reply:
Are you happy for me to resolve this comment @alexcrichton ?